libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/Snackbar.kt

```diff
@@ -0,0 +1,147 @@
+import io.element.android.libraries.theme.ElementTheme
+import io.element.android.libraries.theme.SnackBarLabelColorDark
+import io.element.android.libraries.theme.SnackBarLabelColorLight
+
+@Composable
+fun Snackbar(
+    message: String,
+    modifier: Modifier = Modifier,
+    action: ButtonVisuals? = null,
+    dismissAction: ButtonVisuals? = null,
+    actionOnNewLine: Boolean = false,
+    shape: Shape = RoundedCornerShape(8.dp),
+    containerColor: Color = SnackbarDefaults.color,
+    contentColor: Color = ElementTheme.materialColors.inverseOnSurface,
+    actionContentColor: Color = actionContentColor(),
+    dismissActionContentColor: Color = SnackbarDefaults.dismissActionContentColor,
+) {
+    Snackbar(
+        modifier = modifier,
+        action = action?.let { @Composable { it.Composable() } },
+        dismissAction = dismissAction?.let { @Composable { it.Composable() } },
+        actionOnNewLine = actionOnNewLine,
+        shape = shape,
+        containerColor = containerColor,
+        contentColor = contentColor,
+        actionContentColor = actionContentColor,
+        dismissActionContentColor = dismissActionContentColor,
+        content = { Text(text = message) },
+    )
+}
+
+@Composable
+fun Snackbar(
+    modifier: Modifier = Modifier,
+    action: @Composable (() -> Unit)? = null,
+    dismissAction: @Composable (() -> Unit)? = null,
+    actionOnNewLine: Boolean = false,
+    shape: Shape = RoundedCornerShape(8.dp),
+    containerColor: Color = SnackbarDefaults.color,
+    contentColor: Color = ElementTheme.materialColors.inverseOnSurface,
+    actionContentColor: Color = actionContentColor(),
+    dismissActionContentColor: Color = SnackbarDefaults.dismissActionContentColor,
+    content: @Composable () -> Unit
+) {
+    androidx.compose.material3.Snackbar(
+        modifier = modifier,
+        action = action,
+        dismissAction = dismissAction,
+        actionOnNewLine = actionOnNewLine,
+        shape = shape,
+        containerColor = containerColor,
+        contentColor = contentColor,
+        actionContentColor = actionContentColor,
+        dismissActionContentColor = dismissActionContentColor,
+        content = content,
+    )
+}
+
+// TODO this color is temporary, an `inverse` version should be added to the semantic colors instead
+@Composable
+private fun actionContentColor(): Color {
+    return if (ElementTheme.isLightTheme) {
+        SnackBarLabelColorLight
+    } else {
+        SnackBarLabelColorDark
+    }
+}
+
+@Preview(name = "Snackbar", group = PreviewGroup.Snackbars)
+@Composable
+internal fun SnackbarPreview() {
+    ElementThemedPreview {
+        Snackbar(message = "Snackbar supporting text")
+    }
+}
+
+@Preview(name = "Snackbar with action", group = PreviewGroup.Snackbars)
+@Composable
+internal fun SnackbarWithActionPreview() {
+    ElementThemedPreview {
+        Snackbar(message = "Snackbar supporting text", action = ButtonVisuals.Text("Label", {}))
```

Comment:
Not a big deal, but looks weird in the preview. Instead of "Label", I would have use the word "Action", since this is the text for a button. (applicable to the other occurrences below)

Reply:
I was just trying to match the designs [here](https://www.figma.com/file/G1xy0HDZKJf5TCRFmKb5d5/Compound-Android-Components?type=design&node-id=369%3A37775&mode=design&t=OGE00aBGk5SnHlHN-1), I can revert it.